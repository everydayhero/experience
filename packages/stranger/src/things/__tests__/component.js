import React from 'react'
import renderer from 'react-test-renderer'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import comp from '../component.js'

const shallow = new ReactShallowRenderer()

describe('comp', () => {
  it('should allow removing props from output components to avoid warnings', () => {
    const StyledComp = comp(({ props }) => ({
      color: props.makeit || 'red'
    }))('div')
    const component = renderer.create(
      <StyledComp makeit='green' href='https://everydayhero.com' />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should remove props even when they come from a native component', () => {
    const StyledComp = comp(({ props }) => ({
      color: props.makeit || 'red',
      backgroundColor: 'blue'
    }))('div')
    const StyledCompInehrit = comp(({ props }) => ({
      backgroundColor: props.black || 'black'
    }))(StyledComp)
    const actualShallow = shallow.render(
      <StyledCompInehrit black='white' />
    )
    const actualComponent = renderer.create(
      <StyledCompInehrit black='white' />
    )
    const tree = actualComponent.toJSON()
    expect(actualShallow.props).toMatchSnapshot()
    expect(tree).toMatchSnapshot()
  })
})
