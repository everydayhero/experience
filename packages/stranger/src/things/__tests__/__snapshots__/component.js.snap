exports[`comp should allow removing props from output components to avoid warnings 1`] = `
<div
  className="_1jzdmtb"
  href="https://everydayhero.com" />
`;

exports[`comp should remove props even when they come from a native component 1`] = `
Object {
  "black": "white",
  "className": "_11go4zx",
  "styles": Object {
    "backgroundColor": "white",
  },
}
`;

exports[`comp should remove props even when they come from a native component 2`] = `
<div
  className="_bm5k1h" />
`;
